src/Groups/MatGrp.jl

```diff
@@ -0,0 +1,808 @@
+import AbstractAlgebra: MatElem, MatSpace, parent_type, Ring, RingElem
+import Hecke: base_ring, det, fq_nmod, FqNmodFiniteField, nrows, tr, trace
+import GAP: FFE
+
+export
+    general_linear_group,
+    matrix_group,
+    MatrixGroup,
+    MatrixGroupElem,
+    omega_group,
+    orthogonal_group,
+    special_linear_group,
+    special_orthogonal_group,
+    special_unitary_group,
+    symplectic_group,
+    unitary_group,
+    GL, GO, GU, SL, SO, Sp, SU
+
+
+
+abstract type AbstractMatrixGroupElem <: GAPGroupElem{GAPGroup} end
+
+"""
+    MatrixGroup{RE<:RingElem, T<:MatElem{RE}} <: GAPGroup
+Type of groups `G` of `n x n` matrices over the ring `R`, where `n = degree(G)` and `R = base_ring(G)`.
+
+At the moment, only rings of type `FqNmodFiniteField` are supported.
+"""
+mutable struct MatrixGroup{RE<:RingElem, T<:MatElem{RE}} <: GAPGroup
+   X::GapObj
+   gens::Vector{<:AbstractMatrixGroupElem}
+   ring::Ring
+   deg::Int
+   descr::Symbol                       # e.g. GL
+   ring_iso::GenRingIso
+   mat_iso::GenMatIso
+
+   function MatrixGroup(m::Int64, F::S) where S<:Ring
```

Comment:
```suggestion
   function MatrixGroup(m::Int64, F::Ring) 
```

Reply:
Also, I don't think we support 32-bits platforms, but is there a reason not to use `Int` instead of `Int64` here? (and elsewhere in this PR).